src/components/Inventory.cpp

```diff
@@ -0,0 +1,100 @@
+#include "Inventory.hpp"
+#include <RTTI/RTTI_Inventory.hpp>
+#include <log/logging.hpp>
+
+namespace REGoth
+{
+  Inventory::Inventory(const bs::HSceneObject& parent)
+      : bs::Component(parent)
+  {
+    setName("Inventory");
+  }
+
+  Inventory::~Inventory()
+  {
+  }
+
+  bool Inventory::hasItem(const bs::String& instance) const
+  {
+    throwIfNotUpperCase(instance);
+
+    return mItemCountsByInstance.find(instance) != mItemCountsByInstance.end();
+  }
+
+  bs::UINT32 Inventory::itemCount(const bs::String& instance) const
+  {
+    throwIfNotUpperCase(instance);
+
+    auto it = mItemCountsByInstance.find(instance);
+
+    if (it == mItemCountsByInstance.end()) return 0;
+
+    return it->second;
+  }
+
+  void Inventory::giveItem(const bs::String& instance, bs::UINT32 count)
+  {
+    throwIfNotUpperCase(instance);
+
+    REGOTH_LOG(Info, Uncategorized, "[Inventory] Add {0}x item {1} to Inventory of {2}", count,
+               instance, SO()->getName());
+
+    mItemCountsByInstance[instance] += 1;
```

Comment:
this should read count. 
What happens if we give a count of 0? This has the potential to introduce and entry with the value of 0 which should not exist in mItemCountsByInstance.

Reply:
You're right! I'll take care of that!